### DISCUSSION TITLE
supports to logging apisix_latency and upstream_latency by modifying the full log format

### DISCUSSION THREAD

**Discussion by jagerzhang** [apache/apisix#6010] [Ideas]
I hope apisix can  logging apisix_latency and upstream_latency to help us with request analysis: https://github.com/apache/apisix/issues/5910

At present, it is known that it can be implemented by modifying the full log format with existing logic code (by [kyroslin](https://github.com/kyroslin)): 

https://github.com/jagerzhang/apisix/commit/9c96dba56e9cd851e180c158b4718042a013d841

This should be a very general requirement, Please have a look.

**Comment by tzssangglass** [comment]
cc @spacewander 

From my point of view, this is great. But so far I've only seen you bring it up. Let's wait a little time, how about if more people call for this need and let's activate it?

**Reply by jagerzhang** [reply]
ok

**Comment by jamesislebron** [comment]
can't agree more!
upstream_latency will help a lot for tracing error.

**Comment by spacewander** [comment]
@jagerzhang @jamesislebron 
Could you provide a detailed use case that why this feature is necessary to you?

**Reply by jamesislebron** [reply]
My api users/customers always report that there are some requests wasted a long time; 

With the apisix_latency and upstream_latency log,  I can find out the most time happened on apisix or upstream  conveniently.

**Reply by jagerzhang** [reply]
I heard from the community that access log for apisix also has these two fields, so I think the logs reported by plug-in should also have these fields.

in addition, i think this feature:
- does not affect performance
- references off the exists code
- enriches the log, meet the needs of operation analysis, especially when the user feedback request takes too long

In short, small changes bring big benefits.

**Reply by spacewander** [reply]
I see. So there are strong reasons to add it to the default format. 
PR is welcome!

**Reply by jagerzhang** [reply]
https://github.com/apache/apisix/pull/6063
